### PR TITLE
Q-language docs: Update links and TOCs

### PR DESCRIPTION
**Pull request by Bradben**
- Update the github links to xref links in the existing topics.
- Update TOC entries for new structure
- Removed empty topics from TOC
- Updated redirects for deleted and moved topics

### CODE REVIEW

File: articles/user-guide/host-programs.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ Hence, you can write an operation of the following form:
         }
     }
 ```
-You have defined an operation, `MeasureSuperposition`, which takes no inputs and returns a value of type [Result](https://github.com/microsoft/qsharp-language/tree/main/Specifications/Language/4_TypeSystem#available-types).
+You have defined an operation, `MeasureSuperposition`, which takes no inputs and returns a value of type [Result](xref:microsoft.quantum.qsharp.typesystem-index).
```
Comment (geduardo) [inline articles/user-guide/host-programs.md:51]:
```suggestion
You have defined an operation, `MeasureSuperposition`, which takes no inputs and returns a value of type [Result](xref:microsoft.quantum.qsharp.typesystem-index#available-types).
```
I think we can maintain this, right?

File: articles/user-guide/host-programs.md [hunk 2]
```diff
@@ -158,7 +158,7 @@ Now, a call of `dotnet run` from the command prompt leads to `MeasureSuperpositi
 So, you will see either `One` or `Zero` printed. 
 
 Note that it doesn't matter if you have more callables defined below it, only `MeasureSuperposition` will be run.
-Additionally, it's no problem if your callable includes [documentation comments](https://github.com/microsoft/qsharp-language/blob/main/Specifications/Language/1_ProgramStructure/7_Comments.md#documentation-comments) before its declaration, the `@EntryPoint()` attribute can be simply placed above them.
+Additionally, it's no problem if your callable includes [documentation comments](xref:microsoft.quantum.qsharp.comments) before its declaration, the `@EntryPoint()` attribute can be simply placed above them.
```
Comment (geduardo) [inline articles/user-guide/host-programs.md:161]:
```suggestion
Additionally, it's no problem if your callable includes [documentation comments](xref:microsoft.quantum.qsharp.comments#documentation-comments) before its declaration, the `@EntryPoint()` attribute can be simply placed above them.
```

**Review by geduardo — APPROVED** [review]
Thanks for doing this! I think the only thing left is to add redirects for the deleted files in @bettinaheim's previous PR.

File: articles/user-guide/programs.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ operation MeasureOneQubit() : Result {
 }
 ```
 
-Our [Quantum Katas](https://github.com/microsoft/QuantumKatas#introduction) give a good introduction on [Quantum Computing Concepts](https://github.com/microsoft/QuantumKatas#quantum-computing-concepts-qubits-and-gates) such as common quantum operations and how to manipulate qubits. 
+Our [Quantum Katas](https://.com/microsoft/QuantumKatas#introduction) give a good introduction on [Quantum Computing Concepts](https://github.com/microsoft/QuantumKatas#quantum-computing-concepts-qubits-and-gates) such as common quantum operations and how to manipulate qubits.
```
Comment (geduardo) [inline articles/user-guide/programs.md:56]: I think there's a typo over here

**Review by geduardo — COMMENTED** [review]
Look's good, only a bad link over there

File: articles/user-guide/programs.md (hunk 3 above)
Comment (Bradben) [reply articles/user-guide/programs.md]: Got it, thanks!

**Review by geduardo — APPROVED** [review]

**Pull request merged**